type-definitions/immutable.js.flow

```diff
@@ -27,7 +27,28 @@
  *
  * Note that Immutable values implement the `ESIterable` interface.
  */
-type ESIterable<T> = $Iterable<T,void,void>;
+
+type IteratorResult<Yield,Return> = {
+  done: true,
+  value?: Return,
+} | {
+  done: false,
+  value: Yield,
+};
+
+
+interface _Iterator<Yield,Return,Next> {
+    @@iterator(): _Iterator<Yield,Return,Next>;
+    next(value?: Next): IteratorResult<Yield,Return>;
+}
+
+
+interface _ESIterable<Yield,Return,Next> {
```

Comment:
[see this](https://github.com/facebook/immutable-js/pull/878#discussion_r64115951)